### PR TITLE
Adds the PhysicsNeMo-Mesh changes required for GLOBE 3D

### PR DESCRIPTION
**Pull request by peterdsharpe**
# PhysicsNeMo Pull Request

## Description

Extracts out the PhysicsNeMo-Mesh-related changes required to support GLOBE-3D.

A subset of changes in:
https://github.com/NVIDIA/physicsnemo/pull/1481

Nothing too spicy in here - mostly just minor improvements for torch.compile-friendliness and bugfixes for visualizing low-precision outputs (NumPy doesn't play nice with bf16, for example).

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: physicsnemo/mesh/io/io_pyvista.py [hunk 1]
```diff
@@ -324,7 +324,7 @@ def to_pyvista(
     pv = importlib.import_module("pyvista")
 
     ### Convert points to numpy and pad to 3D if needed (PyVista requires 3D points)
-    points_np = mesh.points.cpu().numpy()
+    points_np = mesh.points.float().cpu().numpy()
```
[the remark this answers is not in the document]
Comment (peterdsharpe) [reply physicsnemo/mesh/io/io_pyvista.py:327]: This is actually required - PyVista will already silently convert to fp32 precision, so we're just making that explicit here.

File: physicsnemo/mesh/geometry/_cell_areas.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        >>> # Regular tetrahedron
+        >>> vecs = torch.tensor([[[1.0, 0.0, 0.0],
+        ...                       [0.5, 0.866025, 0.0],
+        ...                       [0.5, 0.288675, 0.816497]]])
+        >>> compute_cell_areas(vecs).item()  # doctest: +SKIP
+        0.1178...
+    """
+    n_manifold_dims = relative_vectors.shape[-2]
+
+    if n_manifold_dims == 1:
```
Comment (laserkelvin) [inline physicsnemo/mesh/geometry/_cell_areas.py]: This is minor style, but `match/case` looks better IMO for >= 3 cases

File: physicsnemo/mesh/geometry/_cell_areas.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        return _tetrahedron_volumes(relative_vectors)
+    return _gram_det_volumes(relative_vectors)
+
+
+# ---------------------------------------------------------------------------
+# Specialized branches
+# ---------------------------------------------------------------------------
+
+
+def _edge_lengths(
```
Comment (laserkelvin) [inline physicsnemo/mesh/geometry/_cell_areas.py:102]: Nitpicking, but does this really need to be in its own function? I guess it's nice for consistency, but functionally... I'll defer to you to decide how you want to proceed

File: physicsnemo/mesh/io/io_pyvista.py [hunk 4]
```diff
@@ -373,19 +373,19 @@ def to_pyvista(
 
     ### Convert data dictionaries (flatten high-rank tensors for VTK compatibility)
     for k, v in mesh.point_data.items(include_nested=True, leaves_only=True):
-        arr = v.cpu().numpy()
+        arr = v.float().cpu().numpy()
```
Comment (laserkelvin) [inline physicsnemo/mesh/io/io_pyvista.py:376]: Looking at these, I wonder if it might be a bit faster (I don't know if it's critical, or helpful) if we did the `tensordict.float()` (maybe even with `cpu()`), and then within the loop work on the NumPy handles?

**Review by laserkelvin — APPROVED** [review]
Nice refactor! Generally LGTM, and I just added some stylistic suggestions

File: physicsnemo/mesh/geometry/_cell_areas.py (hunk 2 above)
Comment (peterdsharpe) [reply physicsnemo/mesh/geometry/_cell_areas.py]: Haha you're right, that's what the cool kids do :D I'll update

File: physicsnemo/mesh/io/io_pyvista.py (hunk 4 above)
Comment (peterdsharpe) [reply physicsnemo/mesh/io/io_pyvista.py:376]: Oh yeah, good catch! Likely slightly faster, and also slightly better style. Will do.
